### PR TITLE
Pro 4409 cascades rework

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*

Updates the module overview to include a new category - 'Configuration cascades'. Closes PRO-4408

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation
2. Navigate to `/reference/module-api/module-overview.html`

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/reference/module-api/module-overview.md [hunk 1]
```diff
@@ -2,29 +2,20 @@
 
 Module configuration objects may use the following configuration properties. The overall categories are broadly defined:
 - [Configuration settings](#configuration-settings): Static module settings. Once the module is initialized these settings are fixed and can't access the module itself or any other module's settings.
+- [Configuration cascades](#configuration-cascades): Settings that can either be static or set via functions that have access to the module itself.
```
Comment (boutell) [inline docs/reference/module-api/module-overview.md:5]: Practically speaking they have access to the options of the module itself. Access to the rest of `self` is technically there but there's not much attached to it yet.

File: docs/reference/module-api/module-overview.md [hunk 2]
```diff
@@ -127,33 +181,78 @@ module.export = {
...
 - `fields`: An array of field names to include in the group
 
 The `@apostrophecms/doc-type` module arranges the default fields in two groups: `basics` and `utility`. You can override these groups, but those default fields will become ungrouped unless you arrange them again. Any fields not added to a group will be placed in an "Ungrouped" section in the editing interface.
 
+Grouping fields using an object:
+
 ```javascript
 // modules/article/index.js
-module.export = {
+module.exports = {
```
Comment (boutell) [inline docs/reference/module-api/module-overview.md:255]: Oops we did this a lot huh 😆

File: docs/reference/module-api/module-overview.md [hunk 3]
```diff
@@ -315,7 +464,7 @@ The following example uses a hypothetical batch operation that might reset piece
 
 ```javascript
 // modules/article/index.js
-module.export = {
+module.exports = {
   batchOperations: {
     add: {
       // This uses a hypothetical`reset` route added in `apiRoutes`
```
Comment (boutell) [inline docs/reference/module-api/module-overview.md:470]: space

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/module-api/module-overview.md (hunk 1 above)
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:5]: Changed.

File: docs/reference/module-api/module-overview.md [hunk 4]
```diff
@@ -127,33 +181,78 @@ module.export = {
...
+  }
+}
+```
 
 #### `group`
 
 An object of field groups. Groupings are used by the editing interface. Note that `group` _does not apply to widget modules_.
 
-Groups are added as an object with their name as the object key and the following properties:
+Groups are added as an object or a function that returns an object with their name as the object key and the following properties:
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:245]: Changed.

File: docs/reference/module-api/module-overview.md (hunk 2 above)
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:255]: 😸

File: docs/reference/module-api/module-overview.md [hunk 5]
```diff
@@ -171,16 +270,40 @@ module.export = {
...
+    if (options.subtitle) {
+      groupFields.push('subtitle');
+    }
+    return {
+      add: {
+        // ... 
+      },
+      group: {
+        meta: { // 👈 The group's identifying name is the object key.
+        label: 'Article metadata',
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:289]: Fixed.

File: docs/reference/module-api/module-overview.md [hunk 6]
```diff
@@ -171,16 +270,40 @@ module.export = {
...
+  }
+}
+```
 
 ### `filters`
 
 In piece type modules, the `filters` setting configures the pieces manager interface by adding and removing filtering fields (to view only certain pieces). `archived` and `visibility` filters are included by default.
 
-The `filters` object is configured with subsections: `add` and `remove`. Filters must correspond to an existing fields name or custom [query builder](#queries-self-query) on the piece type.
+The `filters` object is configured with subsections: `add` and `remove` and can either be added as a static object or a function that takes `self` and `options` and returns an object. Filters must correspond to an existing fields name or custom [query builder](#queries-self-query) on the piece type.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:302]: Changed.

File: docs/reference/module-api/module-overview.md [hunk 7]
```diff
@@ -243,7 +392,7 @@ An object of columns to add to the piece type manager. Each column is an object
 
 ```javascript
 // modules/article/index.js
-module.export = {
+module.exports = {
   extend: '@apostrophecms/piece-type',
   columns: {
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:397]: Added sentence.

File: docs/reference/module-api/module-overview.md (hunk 3 above)
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:470]: I'm not sure what correction you are expecting here.

File: docs/reference/module-api/module-overview.md [hunk 8]
```diff
@@ -103,21 +94,84 @@ Set to `false` to prevent the module from being fully instantiated in the applic
...
+| [`filters`](#filters) | Object/Function | Configure piece type filters | Piece |
+| [`columns`](#columns) | Object/Function | Configure piece type manager columns | Piece |
+| [`batchOperations`](#batchoperations) | Object/Function | Configure manager batch operations | Piece |
+
+### "Cascading" settings
+These settings can either be configured as a static object or through a function that takes `self` and `options` and returns a configuration object.
+
+As detailed for each setting, the configuration objects have `add`, `remove`, `group`, and `order` properties. This pattern allows these settings to "cascade" from the base classes to project-level classes without requiring those settings be declared again.
+
+Use `add` to add additional settings and `remove` to remove existing base class settings. Use `group` to organize user-facing settings in the editing interface. The `order` option only applies to columns, arranging columns in a particular order.
```
Comment (boutell) [inline docs/reference/module-api/module-overview.md:152]: `order` is useful with any cascade that doesn't need `groups`, which would also include `filters` and `batchOperations`.

**Review by boutell — CHANGES_REQUESTED** [review]
In addition: this is attached to PRO-4409, but that ticket is about documenting the `cascades` property itself, which doesn't seem to happen in this PR, unless I missed it.

I think you probably meant to link it to PRO-4408.

I wouldn't nitpick except that I don't want PRO-4409 to be closed until `cascades` itself is documented. 🙏

File: docs/reference/module-api/module-overview.md [hunk 9]
```diff
@@ -242,9 +242,9 @@ module.exports = {
 
 An object of field groups. Groupings are used by the editing interface. Note that `group` _does not apply to widget modules_.
 
-Groups are added as an object or a function that returns an object with their name as the object key and the following properties:
-- `label`: The visible label (a string) for the group
-- `fields`: An array of field names to include in the group
+`groups` accepts an object composed of named sub-objects. Each sub-object corresponds to a tab in the editing modal, displaying the fields specified within that sub-object. Every sub-object has the following properties:
+- `label`: A string used to label the tab for the group.
+- `fields`: An array of field names to included in the group.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md:247]: Fixed.

File: docs/reference/module-api/module-overview.md (hunk 8 above)
Comment (BoDonkey) [reply docs/reference/module-api/module-overview.md]: Changed/

**Review by boutell — APPROVED** [review]
Good for now, pending the outcome of discussion with Stuart re: batch operation groups

**Pull request merged**